pkg/operator/configobserver/config_observer_controller.go

```diff
@@ -99,10 +102,20 @@ func NewNestedConfigObserver(
 	observers ...ObserveConfigFunc,
 ) factory.Controller {
 	c := &ConfigObserver{
-		operatorClient:   operatorClient,
-		observers:        observers,
-		listers:          listers,
-		nestedConfigPath: nestedConfigPath,
+		operatorClient:        operatorClient,
+		observers:             observers,
+		listers:               listers,
+		nestedConfigPath:      nestedConfigPath,
+		degradedConditionType: condition.ConfigObservationDegradedConditionType,
+	}
+	if len(nestedConfigPath) > 0 {
+		nestedRoot := nestedConfigPath[0]
```

Comment:
put a line here with an example. I have a hard time to follow your printf code and rune logic.